Scripts/RemediationScripts/Remediate-SetClassicFrontDoorMinTLSVersion.ps1

```diff
@@ -0,0 +1,1025 @@
+        Write-Host $([Constants]::SingleDashLine)   
+    }
+
+    # Back up snapshots to `%LocalApplicationData%'.
+    $backupFolderPath = "$([Environment]::GetFolderPath('LocalApplicationData'))\AzTS\Remediation\Subscriptions\$($context.Subscription.SubscriptionId.replace('-','_'))\$($(Get-Date).ToString('yyyyMMddhhmm'))\SetMinTLSVersionForFrontDoors"
+
+    if (-not (Test-Path -Path $backupFolderPath))
+    {
+        New-Item -ItemType Directory -Path $backupFolderPath | Out-Null
+    }
+    
+    # Includes required version of TLS
+    $requiredMinTLSVersion = 1.2
+
+    # Includes Front Door Endpoints where Minimum required TLS version is not set to 1.2.
+    $nonCompliantEndpoints = @()
+
+    # Includes Front Door Endpoints where Minimum required TLS version is either set to 1.2 or blank.
+    $compliantEndpoints = @()
+
+    # Includes Front Doors that were skipped during remediation. There were errors remediating them.
+    $frontDoorEPsSkipped = @()
+
+    $isMinTLSVersionSetOnCustomDomain = @()
+
+    $FrontDoorEndpoints = @()
+
+    $frontDoorWithoutReqMinTLSVersion = @()
+
+    Write-Host "[Step 3 of 4] Fetch all Front Door configurations"
+    Write-Host $([Constants]::SingleDashLine)
+
+    if ([String]::IsNullOrWhiteSpace($FilePath))
+    {
+        $FrontDoorResources | ForEach-Object {
+        $frontDoorResource = $_
+        $resourceId = $_.Id
+        $resourceName = $_.Name
+        $EndpointsWithoutReqTls = @()
+        #$frontendPoints = $_.FrontendEndPoints
+        $resourceGroupName= $_.Id.Split('/')[4]; 
+        try
+        {
+            Write-Host "Fetching Front Door Endpoint configuration: Resource ID: [$($resourceId)], Resource Group Name: [$($resourceGroupName)], Resource Name: [$($resourceName)]..." -ForegroundColor $([Constants]::MessageType.Info)
+            Write-Host $([Constants]::SingleDashLine)
+            
+            $FrontDoorEndpoints = Get-AzFrontDoorFrontendEndpoint -FrontDoorName $resourceName -ResourceGroupName $resourceGroupName
+        
+            if($FrontDoorEndpoints){
+            Write-Host "Front Door Configurations successfully fetched." -ForegroundColor $([Constants]::MessageType.Update)
+            Write-Host $([Constants]::SingleDashLine)
+            }
+        
+        foreach ($item in $FrontDoorEndpoints) 
+        {
+           $minTLSVersionofEndpoint = [decimal]$item.MinimumTlsVersion
+           
+           if(($minTLSVersionofEndpoint | Measure-Object).Count -gt 0)
+           {
+              if($minTLSVersionofEndpoint -lt $requiredMinTLSVersion)
+              {
+                $EndpointsWithoutReqTls += $item 
+                $isMinTLSVersionSetOnCustomDomain = $false;
+              }
+           }
+           else
+           {
+               $compliantEndpoints += $item
+           }   
+        }
+        $totalCount = ($EndpointsWithoutReqTls | Measure-Object).Count
+
+        foreach($item in $EndpointsWithoutReqTls){
+            if (-not $item)
```

Comment:
Is this check required, $item is suposed to be null or fale?

Reply:
fixed as per call